google/cloud/storage/_helpers.py

```diff
@@ -312,3 +402,44 @@ def _convert_to_timestamp(value):
     utc_naive = value.replace(tzinfo=None) - value.utcoffset()
     mtime = (utc_naive - datetime(1970, 1, 1)).total_seconds()
     return mtime
+
+
+def _add_generation_match_parameters(name_value_pairs, **parameters):
+    """Add generation match parameters into the given parameters list.
+
+    :type name_value_pairs: list or dict
+    :param name_value_pairs: Parameters list or dict.
+
+    :type parameters: dict
+    :param parameters: if*generation*match parameters to add.
+    """
+    for snakecase_name, camelcase_name in _GENERATION_MATCH_PARAMETERS:
+        value = parameters.get(snakecase_name)
+
+        if value is not None:
+            if isinstance(name_value_pairs, list):
+                name_value_pairs.append((camelcase_name, value))
+
+            elif isinstance(name_value_pairs, dict):
+                name_value_pairs[camelcase_name] = value
```

Comment:
Added one more condition into helper, as there are methods in which `query_params` is `dict`, not `list`.